Mathlib/Combinatorics/SimpleGraph/Connectivity/Subgraph.lean

```diff
@@ -103,6 +103,22 @@ protected lemma Connected.sup {H K : G.Subgraph}
   rintro ⟨v, (hv|hv)⟩
   · exact Reachable.map (Subgraph.inclusion (le_sup_left : H ≤ H ⊔ K)) (hH ⟨u, hu⟩ ⟨v, hv⟩)
   · exact Reachable.map (Subgraph.inclusion (le_sup_right : K ≤ H ⊔ K)) (hK ⟨u, hu'⟩ ⟨v, hv⟩)
+
+/--
+  This lemma establishes a condition under which a subgraph is the same as a connected component.
+  Note the asymmetry in the hypothesis `h`: `v` is in `H.verts`, but `w` is not required to be.
+-/
+lemma Connected.exists_verts_eq_connectedComponentSupp {H : Subgraph G}
+    (hc : H.Connected) (h : ∀ v ∈ H.verts, ∀ w, G.Adj v w → H.Adj v w) :
```

Comment:
No action required for now but I think worth considering naming this property of `Subgraph`s + building a little API in future work.